src/stores/Schedules.ts

```diff
@@ -0,0 +1,509 @@
+import {
+    amber,
+    blue,
+    blueGrey,
+    cyan,
+    deepPurple,
+    green,
+    indigo,
+    lightGreen,
+    lime,
+    pink,
+    purple,
+    red,
+    teal,
+} from '@material-ui/core/colors';
+
+import { RepeatingCustomEvent } from '../components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog';
+import { CourseInfo, getCourseInfo, queryWebsoc } from '../helpers';
+import { calendarizeCourseEvents, calendarizeCustomEvents, calendarizeFinals } from './calendarizeHelpers';
+import { Schedule, ScheduleCourse, ScheduleSaveState, ScheduleUndoState, ShortCourseSchedule } from './schedule.types';
+
+const arrayOfColors = [
+    red[500],
+    pink[500],
+    purple[500],
+    indigo[500],
+    deepPurple[500],
+    blue[500],
+    green[500],
+    cyan[500],
+    teal[500],
+    lightGreen[500],
+    lime[500],
+    amber[500],
+    blueGrey[500],
+];
+
+export class Schedules {
+    private schedules: Schedule[];
+    private currentScheduleIndex: number;
+    private previousStates: ScheduleUndoState[];
+
+    constructor() {
+        this.schedules = [{ scheduleName: 'Schedule 1', courses: [], customEvents: [] }];
+        this.currentScheduleIndex = 0;
+        this.previousStates = [];
+    }
+
+    // --- Schedule index methods ---
+    /**
+     * @return current schedule index
+     */
+    getCurrentScheduleIndex() {
+        return this.currentScheduleIndex;
+    }
+
+    // --- Schedule related methods ---
+    /**
+     * @return the number of schedules
```

Comment:
Can you remove the autogenerated docstrings if the method name is the same thing as the description?